### PR TITLE
TAG-Observability whitepaper -> Public review

### PR DESCRIPTION
**Pull request by ArthurSens**
### We're in public review phase!

Please take your time to read it, leave your comments, feedback, suggestions, and if you're feeling capable and generous, there is also a contributing section with instructions to what we still want to add to this whitepaper!

If you'd like to add commits to this PR, please open it against my fork: https://github.com/ArthurSens/sig-observability/tree/public-review

After merging your pull request against my fork, your commits will be added to this Pull Request :)

### CODE REVIEW

**Comment by rverchere** [comment]
Hi!

As posted in the slack channel, the CNCF Observability projects include Chaos Engineering tools (see the [Readme section](https://github.com/cncf/tag-observability#cncf-projects-related-to-the-sig)), but there is no references in this whitepaper.

Do you think Chaos Engineering should be mentioned here? Maybe adding some links/relations?

Thanks for your work!

**Comment by ArthurSens** [comment]
> Hi!
> 
> As posted in the slack channel, the CNCF Observability projects include Chaos Engineering tools (see the [Readme section](https://github.com/cncf/tag-observability#cncf-projects-related-to-the-sig)), but there is no references in this whitepaper.
> 
> Do you think Chaos Engineering should be mentioned here? Maybe adding some links/relations?
> 
> Thanks for your work!

Cross-posting what was answered in Slack:

"Hi Rémi, thanks for taking a look at the paper!
From what we've been discussing in the last couple of months, chaos engineering is not a tool that will provide observability data for people who choose to implement it, but Chaos engineering does use observability data(metrics, logs, traces) to provide information. It's more of a consumer than a producer.

To me, it is quite confusing why Chaos Engineering tools were put under TAG-Observability scope.

For example, LitmusChaos is currently under due diligence for reaching "Incubation" maturity and the TAG responsible for due diligence is TAG-app-delivery instead of TAG-Observability 🤔"

maybe @halcyondude, @RichiH or @bwplotka (as the chairs of the TAG) would know why those tools fall under TAG-Observability scope? Do you think we should add be focusing on chaos engineering in the Observability whitepaper?

**Comment by ArthurSens** [comment]
Some extra feedback from slack:

> https://github.com/cncf/tag-observability/blob/main/whitepaper.md#traces
> Do we want to include a screenshot of Jaeger, as it’s application-specific. We are not adding screenshots of Grafana/Prometheus/etc either.
> Additionally, we have a sketch a bit below this, which is more fitting and has basically the same data. Would that be enough?

> https://github.com/cncf/tag-observability/blob/main/whitepaper.md#profiles
> Some of the profilers have capitals, some don’t

> https://github.com/cncf/tag-observability/blob/main/whitepaper.md#achieving-multi-signal-observability
> Typo at Obsevatorium

> https://github.com/cncf/tag-observability/blob/main/whitepaper.md#signal-correlation
>  Examplar is fist introduced on an illustration, but the definition is a bit below that. Should we address this?
> “Powerful logging languages (e.g. LogQL) allow you”, should we include more examples? Also the phrase “logging languages” seems a bit strange. ElasticSearch calls it DSL (domain specific language), Splunk calls it SPL (search processing language), maybe we could use query languages instead?

> https://github.com/cncf/tag-observability/blob/main/whitepaper.md#practical-applications
> “We got an alert about an unexpectedly high error rate exceeding our SLO.” At this point SLO is not introduced. It might be a good idea to link it the the first occurrence.

> https://github.com/cncf/tag-observability/blob/main/whitepaper.md#implementing-slis-slos-and-slas
> The illustration has a typo: “Feeds into product priorization”

> https://github.com/cncf/tag-observability/blob/main/whitepaper.md#multi-signal-correlation
> It starts with “Hopefully, the above write up explains well how to think about observability correlation”, but the section just above this is about Use Cases and SLIs/SLOs, it feels a bit strange

**Comment by prashant3863** [comment]
@ArthurSens Regarding the implementing SLI/O/A excalidraw, I created that and still have the original one. Fixing the typo and raising a PR

**Comment by rverchere** [comment]
References are incorrect:
* Starting from `[1]` to `[7]` in the [references](whitepaper.md#references) section
* Only references to `[4]` (in [introduction](whitepaper.md#introduction)), `[9]` (in [what-is-observability](whitepaper.md#what-is-observability)), `[8]` (in [burn-rate](whitepaper.md#burn-rate))

**Comment by dofinn** [comment]
hi @ArthurSens created a PR against CNCF repo in relation to your request 
https://github.com/cncf/tag-observability/pull/54

**Comment by ArthurSens** [comment]
> References are incorrect:
> 
> * Starting from `[1]` to `[7]` in the [references](whitepaper.md#references) section
> * Only references to `[4]` (in [introduction](whitepaper.md#introduction)), `[9]` (in [what-is-observability](whitepaper.md#what-is-observability)), `[8]` (in [burn-rate](whitepaper.md#burn-rate))

Good catch, when we were working on google docs, the references were "randomly" added. By that I mean that multiple people were working on different things but often using the same reference, some were informing their references, some weren't.

We've tried to do some clean-up and organize the "References" section but looks like we missed some throughout the text.

**Comment by chenmudu** [comment]
Hi, I'm reading.I see that there is no Chinese version of the white paper. If there is no one to translate it, I would like to do it. Could someone give me some advice? Thanks~🌹

**Comment by ArthurSens** [comment]
> Hi, I'm reading.I see that there is no Chinese version of the white paper. If there is no one to translate it, I would like to do it. Could someone give me some advice? Thanks~🌹

Hey, that would be awesome! I'm not sure how to advise you though 😬. I guess you can start translating in a google document and open a PR when ready? Or just create a draft PR and start translating by parts?

Just to let you know, this whitepaper is very much likely to get some small changes in the next weeks. This shouldn't block you in any way, but if you'd like to way to the final version that's alright, too. 🙂

**Comment by chenmudu** [comment]
Cool. Thanks a lot!  Regarding that the content of the whitepaper may change, I think I must do a good job of synchronizing the English version and the Chinese version of the white paper. Then let me start working! @ArthurSens 👍

**Comment by cristinagrosu** [comment]
I was reading through the whitepaper, and I was thinking that it would be interesting to include a chapter to describe potential solutions to reduce the cost of observability. One of them might be to have a smart collector that is able to identify the interesting traces and store only those ones. It seems like something similar was created by Netflix: https://netflixtechblog.com/building-netflixs-distributed-tracing-infrastructure-bb856c319304
Have you considered this as an exploration topic?

**Comment by bwplotka** [comment]
Some discussion during our Obs meeting/feedback / questions from Olga:

1. Difference between Security vs audit log. 
A: Audit log -> subset aspect of security. 

1. System log vs infrastructure log. 
A: There is no difference anymore. Depends on what is a system to you. Could be a cake/layers. It might also not be a technical difference, but rather who is using what.

1. Whitepaper propose naming and conventions. The difference was not very clear to above.

**Comment by bwplotka** [comment]
Hi All, 

Thanks for the feedback and sorry for the lag. As we discussed we volunteered with @RichiH to finalize this work. I *think* I addressed all the comments here and on Slack/other PRs. Let me know if I did not miss anything. 🤗 

> @rverchere: As posted in the slack channel, the CNCF Observability projects include Chaos Engineering tools (see the Readme section), but there is no references in this whitepaper. Do you think Chaos Engineering should be mentioned here? Maybe adding some links/relations?

Chaos Engineering is[ moving to its own chapter](https://docs.google.com/document/d/1scr9uuvG1g1xpIHPs3314FqeFufE31ustTVnRMrX3gI/edit), so the answer is simple here (: 

> Examplar is first introduced on an illustration, but the definition is a bit below that. Should we address this?

Looks like it's still clear for people, so I propose to keep it as it is (: 

> “Powerful logging languages (e.g. LogQL) allow you”, should we include more examples? Also the phrase “logging languages” seems a bit strange. ElasticSearch calls it DSL (domain specific language), Splunk calls it SPL (search processing language), maybe we could use query languages instead?

Fixed! (: However I added only open source examples.

> https://github.com/cncf/tag-observability/blob/main/whitepaper.md#practical-applications
“We got an alert about an unexpectedly high error rate exceeding our SLO.” At this point SLO is not introduced. It might be a good idea to link it the the first occurrence.

Good point, removed mention about SLO for now, since it's not essential.

> References are incorrect:

> Starting from [1] to [7] in the references section
Only references to [4] (in introduction), [9] (in what-is-observability), [8] (in burn-rate)

@rverchere @metonymic-smokey  I moved all the references to actual subsections in the whitepaper and hopefully fixed them all.

>  and I was thinking that it would be interesting to include a chapter to describe potential solutions to reduce the cost of observability. One of them might be to have a smart collector that is able to identify the interesting traces and store only those ones. It seems like something similar was created by Netflix: https://netflixtechblog.com/building-netflixs-distributed-tracing-infrastructure-bb856c319304
Have you considered this as an exploration topic?

Amazing, added your idea to IDEAS section! (:

> Difference between Security vs audit log.
> System log vs infrastructure log.
> Whitepaper propose naming and conventions. The difference was not very clear to above.

Addressed (:

File: whitepaper.md [hunk 1]
```diff
@@ -327,67 +347,70 @@ curl -g 'http://localhost:9090/api/v1/query_exemplars?query=test_exemplar_metric
...
 
 Before we deep dive into the implementation details, we should get the definitions clear as they can be fairly confusing and sometimes be used interchangeably.
 
 * Service Level Indicator (SLI): An SLI is a service level indicator—a carefully defined quantitative measure of some aspect of the level of service that is provided.
-* Service Level Objective (SLO): An SLO is a service level objective: objective for how often you can afford for it to fail. a target value or range of values for a service level that is measured by an SLI
+* Service Level Objective (SLO): An SLO is a service level objective: objective for how often you can afford for it to fail. It is a target value or range of values for a service level that an SLI measures.
 * Service Level Agreement (SLA): a business contract that includes consequences of violating the SLO. This is a targeted percentage
-Error budget: tolerance for failed events over a period of time determined by SLO. This is 100% minus the SLO
+  Error budget: tolerance for failed events over a period of time determined by SLO. This is 100% minus the SLO.
```
Comment (kenfinnigan) [inline whitepaper.md:378]: Should `Error budget` be a definition on same level as SLI/SLO/SLA? i.e. is it missing the `*`?

**Comment by kenfinnigan** [comment]
Can this be closed in favor of https://github.com/cncf/tag-observability/pull/132?

**Comment by bwplotka** [comment]
Yes, sorry for missing this.

**Pull request closed without merging**